lib/modules/dosomething/dosomething_northstar/dosomething_northstar.module

```diff
@@ -158,6 +158,26 @@ function dosomething_northstar_get_northstar_user($drupal_id) {
 }
 
 /**
+ * Get a Northstar user for a given index & type. Useful for non Drupal ID lookups.
+ * @param  string $index_type The type of index to query for (Email, Facebook, Mobile, _id)
+ * @param  string $index The index to query for
+ * @return object|null
+ */
+function dosomething_northstar_get_user_by_custom_index($index_type, $index) {
```

Comment:
Oh you wanted to merge the two, 

> the approach of having less methods and refactor that one a bit 

I must have forgotten this bit while doing everything else. Should be an easy fix, ill just make the index = drupal_id by default so everything in theory continues to work